### DISCUSSION TITLE
optimization problem with expensive and cheap subproblems depending on previous samples

### DISCUSSION THREAD

**Discussion by cgahr** [optuna/optuna#5557] [Q&A] [answered]
Hi @all, 

I want to optimize a function with 2 parameters:
```python
def fun(a, b):
    ...
```
The parameters depend on each other. Basically:
- sampling a new `a` is always expensive
- sampling a new `b` is cheap iff `a` has not been changed compared to the previous sample

Currently, I do a grid search like this:
```python
for a in list_a:
    # first solve step, expensive
    fun2 = partial(fun, a=a)

    for b in list_b:
        # second solve step, cheap (> 1000x)
        value = fun2(b)
```

How would I go about using `optuna` instead, utilizing this structure? Basically I want to sample `a` first and, given `a`, sample 100 different `b`. Both sampled using optuna.

**Comment by nzw0301** [comment, marked as answer]
How about https://optuna.readthedocs.io/en/latest/reference/samplers/generated/optuna.samplers.PartialFixedSampler.html?

**Reply by cgahr** [reply]
I think this might be exactly what I need, didn't even know this existed. Thanks a lot!
